examples/light-switch-app/silabs/src/LightSwitchMgr.cpp

```diff
@@ -159,3 +284,80 @@ void LightSwitchMgr::GenericSwitchWorkerFunction(intptr_t context)
 
     Platform::Delete(data);
 }
+
+void LightSwitchMgr::ButtonEventHandler(uint8_t button, uint8_t btnAction)
+{
+    AppEvent event = {};
+    if (btnAction == to_underlying(SilabsPlatform::ButtonAction::ButtonPressed))
+    {
+        event = LightSwitchMgr::GetInstance().CreateNewEvent(button ? AppEvent::kEventType_UpPressed : AppEvent::kEventType_DownPressed);
+    }
+    else
+    {
+        event = LightSwitchMgr::GetInstance().CreateNewEvent(button ? AppEvent::kEventType_UpReleased : AppEvent::kEventType_DownReleased);
+    }
+    AppTask::GetAppTask().PostEvent(&event);
+}
+
+void LightSwitchMgr::AppEventHandler(AppEvent * aEvent)
+{
+    LightSwitchMgr * lightSwitch = static_cast<LightSwitchMgr *>(aEvent->LightSwitchEvent.Context);
+    switch(aEvent->Type)
+    {
+    case AppEvent::kEventType_ResetWarning:
+        lightSwitch->mResetWarning = true;
+        AppTask::GetAppTask().StartFactoryResetSequence();
+        break;
+    case AppEvent::kEventType_ResetCanceled:
+        lightSwitch->mResetWarning = false;
+        AppTask::GetAppTask().CancelFactoryResetSequence();
+        break;
+    case AppEvent::kEventType_DownPressed:
+        aEvent->Handler = LightSwitchMgr::SwitchActionEventHandler;
+        AppTask::GetAppTask().PostEvent(aEvent);
+        lightSwitch->mDownPressed = true;
+        if (lightSwitch->mLongPressTimer)
+        {
+            lightSwitch->mLongPressTimer->Start();
+        }
+        break;
+    case AppEvent::kEventType_DownReleased:
+        lightSwitch->mDownPressed = false;
+        if (lightSwitch->mLongPressTimer)
+        {
+            lightSwitch->mLongPressTimer->Stop();
+        }
+        if (lightSwitch->mResetWarning)
+        {
+            aEvent->Type = AppEvent::kEventType_ResetCanceled;
+            AppTask::GetAppTask().PostEvent(aEvent);
+        }
+        break;
+    case AppEvent::kEventType_UpPressed:
+    case AppEvent::kEventType_UpReleased:
+        aEvent->Handler = LightSwitchMgr::SwitchActionEventHandler;
+        AppTask::GetAppTask().PostEvent(aEvent);
+        break;
+    default:
+        break;
+    }
+}
+
+void LightSwitchMgr::SwitchActionEventHandler(AppEvent * aEvent)
+{
+    switch(aEvent->Type)
+    {
+    case AppEvent::kEventType_UpPressed:
+        LightSwitchMgr::GetInstance().TriggerLevelControlAction(LevelControl::StepModeEnum::kUp);
+        LightSwitchMgr::GetInstance().GenericSwitchOnInitialPress();
+        break;
+    case AppEvent::kEventType_UpReleased:
+        LightSwitchMgr::GetInstance().GenericSwitchOnShortRelease();
+        break;
```

Comment:
where do you do TriggerLightSwitchAction action now? I think it is missing and if you add it here in UpPressed like to be equivalent of what we are before. You will toggle the lighting everytime you want to raise the level. 


I think the long press logic need to be used for Level Up and Down. 
Like if hold the button (after a small threshold it start moving the level (up or down) Until you release the button.
and when it is a longPress Up release you wouldn't call GenericSwitchOnShortRelease

Reply:
On button press, i'm using the logic of step-with-on-off. So, when the light is at min_level (OnOff value is off) and Button1 is pressed (to increase the level), OnOff value of bounded light app will also be changed. We will not need the TriggerLightSwitchAction explicitly.

Coming to the long press logic, should we configure both the actions when button-0 is long pressed - Factory-reset and level-down?